src/Storages/StorageMySQL.cpp

```diff
@@ -95,8 +95,12 @@ Pipe StorageMySQL::read(
         sample_block.insert({ column_data.type, column_data.name });
     }
 
+    if (!context_->getSettingsRef().external_storage_max_read_rows || max_block_size > context_->getSettingsRef().external_storage_max_read_rows)
```

Comment:
Value of `external_storage_max_read_rows` has non-zero value by default, so it is still not what I meant. Default should be `max_block_size` and with your code it is not so. This check does not do that as default is not zero.

Reply:
In MySQL database engine `max_block_size` controls the ClickHouse clones `n` MySQL rows to a block.

In MaterializeMySQL database engine, before this pr, use  [DEFAULT_BLOCK_SIZE](https://github.com/ClickHouse/ClickHouse/pull/22697/files#diff-585c4235aec5d560e96d2b24faf3ebd4d14dbbbd5d0425e33defda6f4650abfcL331) controls the ClickHouse clone `n` MySQL rows to a block. 

So this will general a question: If rows from an external table are too large(in this case the external table is from MySQL), ClickHouse will be oom killed.

And then, I should add two settings: `external_storage_max_read_rows` to controls the rows number; `external_storage_max_read_bytes` to controls the rows bytes.

In my mind, your means is that we can disable these parameters.
```sql
set external_storage_max_read_rows = 0;
set external_storage_max_read_bytes = 0;
```

But if I disable it the logic in MaterializeMySQL will be modified. In default, every row from MySQL history data will general a block. Because of [this](https://github.com/ClickHouse/ClickHouse/pull/22697/files#diff-33750f6b822a5075a8d8fac60fdd44f214f4906a74a68d157c6f289dd989229eR261).

Looking forward to your reply.